cubi-tools/prototypes/update_workflow.py

```diff
@@ -0,0 +1,646 @@
+    md5_ref = calculate_md5_checksum(workflow_branch_file)
+
+    if md5_local != md5_ref:
+        if dryrun:
+            print(f"The versions of '{file_to_update}' differ!")
+            print(f"Local MD5 checksum: {md5_local}")
+            print(f"Remote MD5 checksum: {md5_ref}")
+            print(f"Update '{file_to_update}'(y/n)? y")
+            print(f"Dry run! '{file_to_update}' would be updated!")
+        else:
+            print(f"The versions of '{file_to_update}' differ!")
+            print(f"Local MD5 checksum: {md5_local}")
+            print(f"Remote MD5 checksum: {md5_ref}")
+            answer_is_pos = user_response(f"Update '{file_to_update}'")
+
+            if answer_is_pos:
+                shutil.copyfile(
+                    workflow_branch.joinpath(file_to_update),
+                    workflow_target.joinpath(file_to_update),
+                )
+                print(f"'{file_to_update}' was updated!")
+            else:
+                print(f"'{file_to_update}' was NOT updated!")
+    else:
+        print(f"'{file_to_update}' is up-to-date!")
+    return None
+
+
+def update_pyproject_toml(workflow_target, workflow_branch, branch, dryrun):
+    """
+    The 'pyproject.toml' is treated a little bit differently. First, there is
+    a check if the file even exists in the project directory. If that is not the
+    case it will be copied into that folder from the desired branch or version tag.
+    If the file is present it will be checked if the cubi.workflow.template.version
+    (and only that information!) differs between the local and the requested branch
+    or version tag version. If that is the case the cubi.workflow.template.version
+    is getting updated. If the 'metadata' switch is set also the cubi.metadata.version
+    will be updated (if necessary).
+    """
+    if not workflow_target.joinpath("pyproject.toml").is_file():
+        if dryrun:
+            print(
+                "\nThere is no 'pyproject.toml' in your folder. "
+                "Do you want to add 'pyproject.toml'(y/n)? y"
+                "\nDry run! 'pyproject.toml' would have been added!"
+            )
+        else:
+            answer_is_pos = user_response(
+                "There is no 'pyproject.toml' in your folder. Add 'pyproject.toml'"
+            )
+
+            if answer_is_pos:
+                shutil.copyfile(
+                    workflow_branch.joinpath("pyproject.toml"),
+                    workflow_target.joinpath("pyproject.toml"),
+                )
+                print("'pyproject.toml' was added!")
+            else:
+                print("'pyproject.toml' was NOT added!")
+    else:
+        pyproject_versions = get_pyproject_versions(workflow_target, workflow_branch)
+        # Just to clearly state which information/files are generated by the
+        # function 'get_metadata_versions(metadata_branch, metadata_target)':
+        branch_metadata_version = pyproject_versions[
+            0
```

Comment:
I assume that somewhat ugly code formatting was generated by black?

Reply:
Yes it is. I try to change it so black won't do it anymore.